ACE/ace/Thread_Manager.cpp

```diff
@@ -1496,14 +1496,14 @@ ACE_Thread_Manager::join (ACE_thread_t tid, ACE_THR_FUNC_RETURN *status)
       {
         if (ACE_OS::thr_equal (biter.next ()->thr_id_, tid))
           {
-            ACE_Thread_Descriptor_Base *tdbl = biter.advance_and_remove (false);
+            ACE_Auto_Ptr<ACE_Thread_Descriptor_Base> tdbl (biter.advance_and_remove (false));
+            ace_mon.release();
```

Comment:
When we have c++11 use a unique_ptr, see similar constructs in ACE

Reply:
Do you mean `ACE_Auto_Ptr` is deprecated?

Use auto ptr similar to https://github.com/DOCGroup/ACE_TAO/commit/8f2464a0b24466f850f36a7f7ad5898c883c67a0

Reply:
Yes, with cpp11 this gives a warning, with cpp17 an error

Reply:
Fixed